### PR TITLE
bpo-44856: Possible reference leak in error paths of update_bases() and __build_class__

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue44856

### CODE REVIEW

**Comment by pablogsal** [comment]
```
❯ ./python -m test test_typing -R :
0:00:00 load avg: 0.54 Run tests sequentially
0:00:00 load avg: 0.54 [1/1] test_typing
beginning 9 repetitions
123456789
.........

== Tests result: SUCCESS ==

1 test OK.

Total duration: 1.0 sec
Tests result: SUCCESS
```

**Comment by pablogsal** [comment]
Funny enough, I tracked this super efficiently with the memory profiler that allows to track Python and C at the same time I am building at work:

example

**Comment by pablogsal** [comment]
This could have been a nightmare to track down since the leak happens in very old code that is quite complex but funny enough, I tracked this super efficiently with the memory profiler that allows to track Python and C at the same time I am building at work:

example

CC: @pradyunsg @lkollar

**Comment by pablogsal** [comment]
This could have been a nightmare to track down since the leak happens in very old code that is quite complex but funny enough, I tracked this super efficiently with the memory profiler that allows to track  **Python and C** at the same time I am building at work with my team:

example

CC: @pradyunsg @lkollar @mariocj89

File: Misc/NEWS.d/next/Core and Builtins/2021-08-07-01-26-12.bpo-44856.9rk3li.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix reference leaks in the error paths of update_bases() and __build_class__. Patch by Pablo Galindo.
```
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2021-08-07-01-26-12.bpo-44856.9rk3li.rst]:
```suggestion
Fix reference leaks in the error paths of ``update_bases()`` and ``__build_class__``. Patch by Pablo Galindo.
```

File: Python/bltinmodule.c [hunk 2]
```diff
@@ -252,13 +243,13 @@ builtin___build_class__(PyObject *self, PyObject *const *args, Py_ssize_t nargs,
     }
 error:
     Py_XDECREF(cell);
-    Py_DECREF(ns);
```
Comment (Fidget-Spinner) [inline Python/bltinmodule.c:255]: Just above this at https://github.com/python/cpython/blob/main/Python/bltinmodule.c#L245,
there seems to be a refleak at the error path too? I'm not too sure, but doesn't it need ``Py_DECREF(cell_cls)`` too? ``PyCell_GET`` returns a new reference.

**Review by Fidget-Spinner — COMMENTED** [review]
FWIW, I tested this on Windows and it also fixes the refleaks. Thanks Pablo, this LGTM.
```
python_d.exe -m test test_typing -R 3:3
0:00:00 Run tests sequentially
0:00:00 [1/1] test_typing
beginning 6 repetitions
123456
......

== Tests result: SUCCESS ==

1 test OK.
```

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Pull request merged**